src/main/kotlin/com/example/olaf/blogtutorial/HtmlController.kt

```diff
@@ -1,15 +1,48 @@
 package com.example.olaf.blogtutorial
 
+import org.springframework.http.HttpStatus
 import org.springframework.stereotype.Controller
 import org.springframework.ui.Model
 import org.springframework.ui.set
 import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.server.ResponseStatusException
 
 @Controller
-class HtmlController {
+class HtmlController(private val repository: ArticleRepository) {
     @GetMapping("/")
     fun blog(model: Model): String {
         model["title"] = "Blog"
+        model["articles"] = repository.findAllByOrderByAddedAtDesc().map { it.render() }
         return "blog"
     }
+
+    @GetMapping("/article/{slug}")
+    fun article(@PathVariable slug: String, model: Model): String {
+        val article = repository
+            .findBySlug(slug)
+            ?.render()
+            ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "This article does not exist")
+        model["title"] = article.title
+        model["article"] = article
```

Comment:
```suggestion
        model["article"] = article
```
title 은 아티클 안에 있는 값 같은데 이렇게 2개를 주면 중복 같아요.
하나만 내리고,
뷰에서 article.title 로 쓰면 될 것 같아요.

Reply:
넵 수정했습니다 👍